.github/workflows/publish-api-doc.yml

```diff
@@ -0,0 +1,63 @@
+
+# This generates Doxygen autodocs for the C++ API (for the libyui base lib)
+# and pushes them to the "gh-pages" branch which publishes them
+# to the project's GitHub pages at https://libyui.github.io/libyui/api-doc/
+#
+# See also
+#
+#   https://gist.github.com/shundhammer/ed359db0d9329d4db551528256060d2a
+#
+# GitHub actions syntax:
+#   https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions
+
+name: publish-api-doc
+
+# Trigger this when a pull request is merged (which implies pushing to master).
+on:
+  push:
+    branches:
+      # master
+      - main
+
+      # push to this branch to trigger a test run of this action
+      # (which does the real publishing, but you don't need to merge a PR)
+      - api-doc-test
```

Comment:
You can re-trigger the action manually from https://github.com/libyui/libyui/actions, or we could use the [repository_dispatch](https://docs.github.com/en/actions/reference/events-that-trigger-workflows#repository_dispatch) event [triggered manually](https://docs.github.com/en/rest/reference/repos#create-a-repository-dispatch-event).

This looks a bit fragile to me, if you commit anything else than the `master` branch there (by mistake) then it could publish the docu for a wrong branch.

Reply:
OK. Removed that branch from here.